RequestKit/Router.swift

```diff
@@ -31,61 +31,73 @@ public protocol Router {
     var method: HTTPMethod { get }
     var path: String { get }
     var encoding: HTTPEncoding { get }
-    var params: [String: String] { get }
+    var params: [String: AnyObject] { get }
     var configuration: Configuration { get }
 
-    func urlQuery(parameters: [String: String]) -> String
-    func request(urlString: String, parameters: [String: String]) -> NSURLRequest?
+    func urlQuery(parameters: [String: AnyObject]) -> [NSURLQueryItem]
+    func request(urlComponents: NSURLComponents, parameters: [String: AnyObject]) -> NSURLRequest?
     func loadJSON<T>(session: RequestKitURLSession, expectedResultType: T.Type, completion: (json: T?, error: ErrorType?) -> Void) -> URLSessionDataTaskProtocol?
     func request() -> NSURLRequest?
 }
 
 public extension Router {
     public func request() -> NSURLRequest? {
-        let URLString = configuration.apiEndpoint.stringByAppendingURLPath(path)
+        let url = NSURL(string: path, relativeToURL: NSURL(string: configuration.apiEndpoint))
         var parameters = encoding == .JSON ? [:] : params
         if let accessToken = configuration.accessToken {
             parameters[configuration.accessTokenFieldName] = accessToken
         }
-        return request(URLString, parameters: parameters)
+        let components = NSURLComponents(URL: url!, resolvingAgainstBaseURL: true)
+        return request(components!, parameters: parameters)
     }
 
-    public func urlQuery(parameters: [String: String]) -> String {
-        var components: [(String, String)] = []
+    public func urlQuery(parameters: [String: AnyObject]) -> [NSURLQueryItem] {
+        var components: [NSURLQueryItem] = []
         for key in parameters.keys.sort(<) {
-            if let value = parameters[key] {
-                let encodedValue = value.urlEncodedString()
-                components.append(key, encodedValue!)
+            guard let value = parameters[key] else { continue }
+            switch value {
+            case let value as String:
+                if let escapedValue = value.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLQueryAllowedCharacterSet()) {
+                    components.append(NSURLQueryItem(name: key, value: escapedValue))
+                }
+            case let valueArray as [String]:
+                valueArray.forEach { item in
+                    if let escapedValue = item.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLQueryAllowedCharacterSet()) {
+                        components.append(NSURLQueryItem(name: "\(key)[\(valueArray.indexOf(item)!)]", value: escapedValue))
```

Comment:
Instead of calling indexOf inside a forEach closure, use a simple for loop over the range `0..<valueArray.count`.